dlib/dnn/loss.h

```diff
@@ -1142,6 +1143,8 @@ namespace dlib
                 std::vector<size_t> truth_idxs;
                 truth_idxs.reserve(truth->size());
 
+                std::unordered_map<size_t, std::deque<rectangle>> truth_idx_to_truth_rects;
```

Comment:
Seems like this could be simply `std::unordered_map<size_t, rectangle>` since only one element of the deque is used.  That would also likely execute faster, which is worth it here as this code can sometimes be a bottleneck.

Reply:
Changed.